### PR TITLE
[WIP] Implement Capi checker

### PR DESCRIPTION
**Pull request by ashish-amarnath**
**What this PR does / why we need it**:
Builds a debugging tool to run checks on capi cluster/ environment

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
NONE
```

/assign @chuckha 
Accepting renaming suggestions for the tool 😄

### CODE REVIEW

**Comment by ashish-amarnath** [comment]
E.g. output
```
💰 ./bin/capi-checker -check-yaml ~/Desktop/check-config.yaml
Running "capi-checker" version info:
GitReleaseTag: "UNKNOWN_RELEASE", Major: "UNKNOWN_GIT_MAJOR_VERSION", Minor: "UNKNOWN_GIT_MINOR_VERSION", GitRelaseCommit: "UNKNOWN_RELEAE_COMMIT"
Git Branch: "capi-checker"
Git commit: "7f1313f75c66b6"
Git tree state: "clean"
Using kubeconfig "/Users/ashisha/.kube/kind-config-management" from env var "KUBECONFIG"
Reading check config from "/Users/ashisha/Desktop/check-config.yaml"
Running checks "foo"
🧪  Checking for [kind: k8s-statefulset, namespace: system, name: controller-manager]...
	 ✅    Result: gcr.io/k8s-cluster-api/cluster-api-controller:v0.1.4
🧪  Checking for [kind: k8s-statefulset, namespace: docker-provider-system, name: docker-provider-controller-manager]...
	 ✅    Result: gcr.io/kubernetes1-226021/capd-manager:latest
🧪  Checking for [kind: k8s-deploy, namespace: docker-provider-system, name: docker-provider-controller-manager]...
	 🛑    Failed, Reason="deployments.extensions \"docker-provider-controller-manager\" not found"

```

**Comment by ashish-amarnath** [comment]
Something is up w/ the `hack/verify-gotest.sh`
Passes locally

File: cmd/capi-checker/main.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	"io/ioutil"
+	"os"
+
+	"sigs.k8s.io/cluster-api-provider-docker/cmd/capi-checker/pkg/check"
+	"sigs.k8s.io/cluster-api-provider-docker/cmd/versioninfo"
+)
+
+// Constants
+const (
+	KUBECONFIG = "KUBECONFIG"
```
Comment (chuckha) [inline cmd/capi-checker/main.go]: the convention in go is to use use either an exported variable like `Kubeconfig` or a local-to-package variable like `kubeconfig`. Depending on what you want (probably `kubeconfig`) please use one of the two instead.

File: cmd/capi-checker/main.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	}
+
+	conf, err := check.ConfigFromYaml(data)
+	if err != nil {
+		fmt.Printf("Invalid yaml file %q\n", *checkConfig)
+		os.Exit(301)
+	}
+
+	fmt.Printf("Running checks %q\n", conf.Name)
+	for _, check := range conf.Checks {
```
Comment (chuckha) [inline cmd/capi-checker/main.go]: consider https://golang.org/pkg/text/tabwriter/

**Comment by chuckha** [comment]
Oh I think this is related to the versioning stuff.

Do you think we can keep that to something like, a kubectl command + grep for now? This feels like it's generalizing too early and I don't know what other utility we'd see out of it

**Comment by ashish-amarnath** [comment]
Ye, it is related to the versioning issue.
To start off, the idea was to build a shareable checker tool to assist
developers in debugging the environment- to know what versions of capi and
providers are being used and check/ report compatibility between versions.

I think this can also be extended to build some sort of conformance
checking.

**Comment by ashish-amarnath** [comment]
https://github.com/kubernetes-sigs/cluster-api-provider-docker/issues/88#issuecomment-512290236

**Comment by chuckha** [comment]
ah yes, that comment was just me having a general idea, but as stated it is very out of scope for this issue. We should keep this simple and if you want to build something like that outside of this repo that seems perfectly fine to me.

**Comment by ashish-amarnath** [comment]
Will move this tool to some other place. Closing this PR

**Pull request closed without merging**